nemo_text_processing/g2p/models/ctc_g2p.py

```diff
@@ -0,0 +1,417 @@
+    #     return {
+    #         "input_ids": NeuralType(('B', 'T'), TokenIndex()),
+    #         "attention_mask": NeuralType(('B', 'T'), MaskType(), optional=True),
+    #         "labels": NeuralType(('B', 'T'), LabelsType()),
+    #     }
+    #
+    # @property
+    # def output_types(self) -> Optional[Dict[str, NeuralType]]:
+    #     return {"loss": NeuralType((), LossType())}
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        self.world_size = 1
+        if trainer is not None:
+            self.world_size = trainer.num_nodes * trainer.num_gpus
+
+        self.mode = cfg.model_name.lower()
+
+        supported_modes = ["byt5", "conformer_bpe"]
+        if self.mode not in supported_modes:
+            raise ValueError(f"{self.mode} is not supported, choose from {supported_modes}")
+
+        # Setup phoneme tokenizer
+        self._setup_tokenizer(cfg.tokenizer)
+
+        # Setup grapheme tokenizer
+        self.tokenizer_grapheme = self.setup_grapheme_tokenizer(cfg)
+
+        # Initialize vocabulary
+        vocabulary = self.tokenizer.tokenizer.get_vocab()
+        cfg.decoder.vocabulary = ListConfig(list(vocabulary.keys()))
+        self.vocabulary = cfg.decoder.vocabulary
+        self.labels_tkn2id = {l: i for i, l in enumerate(self.vocabulary)}
+        self.labels_id2tkn = {i: l for i, l in enumerate(self.vocabulary)}
+
+        super().__init__(cfg, trainer)
+
+        self._setup_encoder()
+        self.decoder = EncDecCTCModel.from_config_dict(self._cfg.decoder)
+        self.loss = CTCLoss(
+            num_classes=self.decoder.num_classes_with_blank - 1,
+            zero_infinity=True,
+            reduction=self._cfg.get("ctc_reduction", "mean_batch"),
+        )
+
+        # Setup decoding objects
+        decoding_cfg = self.cfg.get('decoding', None)
+
+        # In case decoding config not found, use default config
+        if decoding_cfg is None:
+            decoding_cfg = OmegaConf.structured(CTCBPEDecodingConfig)
+            with open_dict(self.cfg):
+                self.cfg.decoding = decoding_cfg
+
+        self.decoding = CTCBPEDecoding(self.cfg.decoding, tokenizer=self.tokenizer)
+
+        self._wer = WERBPE(decoding=self.decoding, use_cer=False, log_prediction=False, dist_sync_on_step=True,)
+        self._per = WERBPE(decoding=self.decoding, use_cer=True, log_prediction=False, dist_sync_on_step=True,)
+
+    def setup_grapheme_tokenizer(self, cfg):
+        """ Initialized grapheme tokenizer """
+
+        if self.mode == "byt5":
+            # Load appropriate tokenizer from HuggingFace
+            grapheme_tokenizer = AutoTokenizer.from_pretrained(cfg.tokenizer_grapheme.pretrained)
+            self.max_source_len = cfg.get("max_source_len", grapheme_tokenizer.model_max_length)
+            self.max_target_len = cfg.get("max_target_len", grapheme_tokenizer.model_max_length)
+
+            # TODO store byt5 vocab file
+        else:
```

Comment:
Can we make the `conformer_bpe` case explicit to help readability and just in case something weird happens in between the check in init?

Reply:
done